src/sentry/web/frontend/base.py

```diff
@@ -377,8 +378,10 @@ def handle_permission_required(self, request: Request, organization, *args, **kw
                 "access.must-sso",
                 extra={"organization_id": organization.id, "user_id": request.user.id},
             )
-            auth.initiate_login(request, next_url=request.get_full_path())
-            redirect_uri = reverse("sentry-auth-organization", args=[organization.slug])
+            auth.initiate_login(request)
+            path = reverse("sentry-auth-organization", args=[organization.slug])
+            redirect_uri = make_login_link_with_redirect(path, request.get_full_path())
```

Comment:
sets `redirect_uri` to the previous path before they were logged out.

Reply:
not exactly, here redirect_uri will be the organization login, with the previous path appended as a query parameter

Reply:
yeah sorry thats what i meant, `make_login_link_with_redirect` takes `path` which is the org's login and `make_login_link_with_redirect` which is the previous path

Reply:
users still may manually modify the `next` query parameter themselves to an external URL, but we'll never redirect to it. do we think it's worth modifying our base login views to sanitize this parameter anyways to be safe? (think behavior would be just strip the query parameter and redirect to the login page they were trying to access).

Reply:
i've looked at how some other apps handle this, and i don't think we need to sanitize the parameter on every GET.